sdk/identity/src/token_credentials/client_secret_credentials.rs

```diff
@@ -59,39 +26,33 @@ pub mod tenant_ids {
 #[derive(Debug)]
 pub struct ClientSecretCredential {
     http_client: Arc<dyn HttpClient>,
+    authority_host: Url,
     tenant_id: String,
     client_id: oauth2::ClientId,
     client_secret: Option<oauth2::ClientSecret>,
-    options: TokenCredentialOptions,
     cache: TokenCache,
 }
 
 impl ClientSecretCredential {
     /// Create a new `ClientSecretCredential`
     pub fn new(
-        http_client: Arc<dyn HttpClient>,
+        options: TokenCredentialOptions,
```

Comment:
`impl Into<TokenCredentialOptions>`. Adding a FROM<Arc<dyn HttpClient>` would make this non-breaking.